### DISCUSSION TITLE
Labelに値が反映されない

### DISCUSSION THREAD

**Discussion by mm0o21** [lit-iphone-course/lit-iphone-course#71] [Question-original]
(1) 実現したいこと
ピンを追加した後、ピンをタップした時に表示されるモーダルにrealmに保存してある住所、日付を表示したい
(2) 実現するために自分が試した内容とその結果
ピンを押した時に画面を遷移させ、押したピンの情報を引っ張ってくるということができず困っています＞＜
(3) 調べた内容や資料
[https://qiita.com/ryoegu/items/2249555be43dc7ed617a](url)
(4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)
```swift
override func viewWillAppear(_ animated: Bool) {
        presentingViewController?.beginAppearanceTransition(false, animated: animated)
            super.viewWillAppear(animated)

            adrLabel.text = adr

            DispatchQueue(label: "background").async {
                let realm = try! Realm()

                if realm.objects(Pin.self).last != nil {
                    DispatchQueue.main.async {
                        self.adrLabel.text = self.adr
                        self.dateField.text = self.date
                        print("ここだよーーーーー")
                    }
                }
            }

        }
```

これでいけると思っていたのですが、画面遷移するだけで、adrもdateもnilになってしまいます。。
(5) GitHubのリンク
[https://github.com/mm0o21/sanphoto](url)

**Comment by Yonpachi48** [comment]
realmに追加する時点でプライマリーキーであるidも指定して保存しましょう。これは指定したpin特有のものにしたほうがいいので緯度経度とかにするといいかも！
次にそのプライマリーキーを用いてrealmの情報を取得します。やり方は
> if let pin = realm.object(ofType: Pin.self, forPrimaryKey: 1)  { 
>       self.dateField.text = pin.date
>  }

こんな感じになります。
→[参考サイト](https://qiita.com/See_Ku/items/6211690df3c56255fde4)

あと、realmは異なるスレッドを跨いでアクセスするとエラーになるので一つのスレッドの中に収めたほうがよき！
→[参考サイト](https://qiita.com/reo0612/items/8a373fdd2b9294773550)
